reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/RoleResource.java

```diff
@@ -244,6 +244,46 @@ public RoleResource(RoleRepository roleRepository, RoleMapper roleMapper,
     return new ResponseEntity<>(result, HttpStatus.OK);
   }
 
+  /**
+   * EXISTS /roles/matches/ : check if there's a role in the database that matches the code
+   * provided, regardless of casing.
+   *
+   * @param codes            the codes of the RoleDTOs to check
+   * @param columnFilterJson The column filters to apply
+   * @return Map             Where a key is the code to be matched, and a value is the code that was
+   *     matched from the database.
+   */
+  @PostMapping("/roles/matches/")
+  public ResponseEntity<Map<String, String>> rolesMatch(@RequestBody List<String> codes,
+      @RequestParam(value = "columnFilters", required = false) String columnFilterJson)
+      throws IOException {
+    codes = codes.stream()
+        .map(code -> getConverter(code).decodeUrl().toString())
+        .collect(Collectors.toList());
+    log.debug("REST request to check Roles match: {}", codes);
```

Comment:
Its ok here because its debug logging only, but in general this can be vulnerable to log injection - we had to do some clean-ups on some of the recent projects we open-sourced to guard against this where the log level was 'info' and URL parameters were being logged as-is.

Reply:
So I'll leave it in?

Reply:
Yeah, its fine - it was just a comment for interest :)